code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm

```diff
@@ -0,0 +1,1195 @@
+				continue
+			if(alltep)
+				new /mob/living/simple_mob/vore/alienanimals/teppi/baby(loc, src, alltep)
+				baby_countdown = 200
+				if(affinity[alltep.real_name])
+					return
+				handle_affinity(alltep, 30) //Mom and dad should like eachother when they do their business
+				alltep.handle_affinity(src, 30)
+				return
+/mob/living/simple_mob/vore/alienanimals/teppi/proc/teppi_sound()
+	if(!teppi_adult || client)
+		return
+	if(resting)
+		return
+	playsound(src, pick(teppi_sound), 75, 1)
+
+//Handles both growing up from a baby and also passing parent details to new babies.
+/mob/living/simple_mob/vore/alienanimals/teppi/New(newloc, teppi1, teppi2)
+	GLOB.teppi_count += 1
+	if(teppi1 && !teppi2)
+		inherit_from_baby(teppi1)
+	else if (teppi1 && teppi2)
+		inherit_from_parents(teppi1, teppi2)
+	..()
+
+/mob/living/simple_mob/vore/alienanimals/teppi/Destroy()
+	GLOB.teppi_count -= 1
+	friend_zone = null
+	active_ghost_pods -= src
+	ai_holder.leader = null
+	return ..()
+
+/mob/living/simple_mob/vore/alienanimals/teppi/Login()
+	..()
+	to_chat(src, "<span class='notice'>While you may have access to galactic common, this is purely meant for making it so you can understand people in an OOC manner, for facilitating roleplay. You almost certainly should not be speaking to people or roleplaying as though you understand everything everyone says perfectly, but it's not unreasonable to be able to intuit intent and such through people's tones when they speak. Teppi are kind of smart, but they are animals, and should be roleplayed as such.</span> <span class='warning'>ADDITIONALLY, you have the ability to produce offspring if you're well fed enough every once in a while, and the ability to disable this from happening to you. These verbs exist for to preserve the mechanical functionality of the mob you are playing. You should be aware of your surroundings when you use this verb, and NEVER use it to prefbreak or be disruptive. If in doubt, don't use it.</span> <span class='notice'>Also, to note, AI Teppi will never initiate breeding with player Teppi.</span>")
+
+/mob/living/simple_mob/vore/alienanimals/teppi/lay_down()
+	..()
+	if(client || !teppi_adult)
+		return
+	if(vore_selected == friend_zone)
+		return
+	if(resting)
+		vore_selected.digestchance = 60
+		vore_selected.digest_brute = 6
+		vore_selected.digest_burn = 6
+	else
+		vore_selected.digestchance = 5
+		vore_selected.digest_brute = 0.05
+		vore_selected.digest_burn = 0.05
+
+/mob/living/simple_mob/vore/alienanimals/teppi/animal_nom(mob/living/T in living_mobs(1))
+	if(client)
+		return ..()
+	var/current_affinity = affinity[T.real_name]
+	ai_holder.busy = TRUE
+	T.stop_pulling()
+	if(current_affinity >= 50)
+		var/tumby = vore_selected
+		vore_selected = friend_zone
+		..()
+		vore_selected = tumby
+		return
+	else if(current_affinity <= -50)
+		vore_selected.digest_mode = DM_DIGEST
+	else 
+		vore_selected.digest_mode = DM_DRAIN
+	..()
+	ai_holder.busy = FALSE
```

Comment:
Comment: This seems particularly annoying to have to do this way. Maybe in the future there can be a better way to just pick a belly to use.